### PR TITLE
StaticJavaParser and SourceRoot APIs #3415

### PR DESCRIPTION
**Pull request by nelson-ng-96**
Marked old method in StaticJavaParser as deprecated and created a new method that aligns with the SourceRoot method

### CODE REVIEW

File: javaparser-core/src/main/java/com/github/javaparser/StaticJavaParser.java [hunk 1]
```diff
@@ -64,13 +64,20 @@ private StaticJavaParser() {
     }
 
     /**
-     * Get the configuration for the parse... methods.
+     * Get the configuration for the parse... methods. Deprecated method.
      */
     @Deprecated
     public static ParserConfiguration getConfiguration() {
         return localConfiguration.get();
```
Comment (TheJosephCCox) [inline javaparser-core/src/main/java/com/github/javaparser/StaticJavaParser.java:71]: I think they just want you to call the new method, not leave any code here (replace with `return getParserConfiguration()`). Not sure on this though, I guess they will comment on PR.

File: javaparser-core/src/main/java/com/github/javaparser/StaticJavaParser.java [hunk 2]
```diff
@@ -64,13 +64,20 @@ private StaticJavaParser() {
...
     }
 
+    /**
+     * Get the configuration for the parse... methods.
+     */
+    public static ParserConfiguration getParseConfiguration() {
```
[the remark this answers is not in the document]
Comment (nelson-ng-96) [reply javaparser-core/src/main/java/com/github/javaparser/StaticJavaParser.java:77]: I fixed this typo.

File: javaparser-core/src/main/java/com/github/javaparser/StaticJavaParser.java (hunk 1 above)
Comment (nelson-ng-96) [reply javaparser-core/src/main/java/com/github/javaparser/StaticJavaParser.java:71]: I added the new return value for the deprecated method.

**Comment by TheJosephCCox** [comment]
Looks good

**Comment by flanbino** [comment]
Looks fine to me too 👍

**Review by flanbino — APPROVED** [review]

**Comment by Bash-09** [comment]
I don't see any issues with these changes.

**Pull request closed without merging**